electrum/dns_hacks.py

```diff
@@ -69,8 +62,15 @@ def resolve_with_dnspython(host):
         addrs = []
         expected_errors = (dns.resolver.NXDOMAIN, dns.resolver.NoAnswer,
                            concurrent.futures.CancelledError, concurrent.futures.TimeoutError)
-        ipv6_fut = _dns_threads_executor.submit(dns.resolver.resolve, host, dns.rdatatype.AAAA)
-        ipv4_fut = _dns_threads_executor.submit(dns.resolver.resolve, host, dns.rdatatype.A)
+        loop = get_asyncio_loop()
+        ipv6_fut = asyncio.run_coroutine_threadsafe(
+            dns.asyncresolver.resolve(host, dns.rdatatype.AAAA),
+            loop,
+        )
+        ipv4_fut = asyncio.run_coroutine_threadsafe(
```

Comment:
Note: calling `.result()` on these futures will deadlock if we are running on the asyncio thread - which was not the case before.
Though socket.getaddrinfo is unlikely to run on the asyncio thread I think.

Reply:
I added an assert in https://github.com/spesmilo/electrum/commit/e75476430ca50a3c572b7cb55fcf8a83e668e98a